File: MVCDemo-OC/MVCDemo-OC/MyCell.m
//
//  MyCell.m
//  MVCDemo-OC
//
//  Created by xyxxllh on 16/7/27.
//  Copyright © 2016年 XunLei. All rights reserved.
//

#import "MyCell.h"
#import "Header.h"
#import "CellModel.h"

@interface MyCell()
@property (weak, nonatomic) IBOutlet UILabel *hitLabel;

@end

@implementation MyCell

- (BOOL)update:(id<TTModel>)model{
    if ([model isKindOfClass:[CellModel class]]) {
        CellModel *m = model;
        self.hitLabel.text = [m hit];
        return YES;
    }
    return NO;
}

@end

File: MVCDemo-swift/MVCDemo-swift/MyCell.swift
//
//  MyCell.swift
//  MVCDemo-swift
//
//  Created by xyxxllh on 16/7/27.
//  Copyright © 2016年 XunLei. All rights reserved.
//

import UIKit

class MyCell: UITableViewCell {
    
    @IBOutlet weak var hitLable: UILabel!
    
    override func awakeFromNib() {
        layoutMargins = UIEdgeInsetsZero
    }
}

extension MyCell : TTView {
    func update<T : TTModel>(model: T) -> Bool{
        guard model is CellModel else {
            return false
        }
        
        let m =  model as! CellModel
        
        hitLable.text = m.hit
        
        return true
    }
}
